src/auth/oauth2/CMakeLists.txt

```diff
@@ -112,19 +98,11 @@ target_compile_definitions(authmethod_oauth2_a PRIVATE "CMAKE_SOURCE_DIR=\"${CMA
 
 target_link_libraries(authmethod_oauth2_a qgis_core)
 
-if(WITH_INTERNAL_O2)
-  target_include_directories(authmethod_oauth2_a SYSTEM PUBLIC ${O2_INCLUDE_DIR})
-else()
-  if(NOT "${O2_LIBRARY}" STREQUAL "")
-    # prefer dynamic linking
-    target_link_libraries(authmethod_oauth2_a ${O2_LIBRARY})
-  else()
-    target_link_libraries(authmethod_oauth2_a ${O2_LIBRARY_STATIC})
-  endif()
-endif()
+target_include_directories(authmethod_oauth2_a SYSTEM PUBLIC ${O2_INCLUDE_DIR})
 
 target_include_directories(authmethod_oauth2_a PRIVATE
   ${CMAKE_SOURCE_DIR}/external/qjsonwrapper
+  ${CMAKE_SOURCE_DIR}/external/o2/src
```

Comment:
With this change we accept the fact that we are adopting a forked version of o2.
This makes it different from other external "convenience" libraries to overcome the lack of these in some distributions.

I would suggest we move this out of the `external` folder into `src/auth/oauth2`, as all the cmake logic to build this library also sits in here.
Ideally I would also suggest to namespace the code to avoid symbol collision when linking to other libraries that depend on an upstream version of `o2`. (I've gone through some pain for poly2tri and similar libs in the past).

Reply:
@m-kuhn 
> With this change we accept the fact that we are adopting a forked version of o2.

I don't see any change in the situation here. This was all dead code, unused by anyway since the introduction of the method.

Instead of abandoning any hope of a maintained upstream, I'd prefer to leave things as they are BUT fork the library repo to QGIS/o2, and then maintain our patchset on there. At least then we'd be making a semi-maintained version of the library available for others to use...